lightning/impl/sgd_fast.pyx

```diff
@@ -53,6 +53,21 @@ cdef class ModifiedHuber(LossFunction):
         else:
             return 4.0 * y
 
+cdef class WeightedLoss(LossFunction):
+
+    cdef double[:] sample_weights
+    cdef LossFunction loss_func
+
+    def __init__(self, np.ndarray[double, ndim=1] sample_weights, LossFunction loss_func):
+        self.sample_weights = sample_weights
+        self.loss_func = loss_func
+
+    cpdef double loss(self, double p, double y, int i):
+        return self.loss_func.loss(p, y, i) * self.sample_weights[i]
+
+    cpdef double get_update(self, double p, double y, int i):
+        return self.loss_func.get_update(p, y, i) * self.sample_weights[i]
```

Comment:
Instead of using `WeightedLoss`, have you tried added `sample_weights` directly to `Loss`? I know that this will cause an addition multiplication per sample but I would be curious to know the performance hit.